core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -895,6 +898,38 @@ private RelNode mockCountOver(String table,
     assertThat(s, isLinux(expected));
   }
 
+  @Test void testTableModify() {
+    final FrameworkConfig config = RelBuilderTest.config().build();
+    final RelBuilder builder = RelBuilder.create(config);
+    final RexBuilder rexBuilder = builder.getRexBuilder();
+    LogicalTableScan scan = (LogicalTableScan) builder.scan("EMP").build();
+    List<RexNode> projExprs = new ArrayList<>();
+    int inputFieldNum = scan.getRowType().getFieldCount();
+    for (int i = 0; i < inputFieldNum; i++) {
+      RexNode projExpr = rexBuilder.makeInputRef(scan, i);
+      projExprs.add(projExpr);
+    }
+    LogicalProject project = LogicalProject.create(
+        scan, ImmutableList.of(), projExprs, scan.getRowType());
+    LogicalTableModify modify = LogicalTableModify.create(
+        scan.getTable(),
+        (Prepare.CatalogReader) scan.getTable().getRelOptSchema(),
+        project,
+        TableModify.Operation.INSERT,
+        null,
```

Comment:
All the operations should be tested, i.e. the `DELETE`, `UPDATE` and `MERGE`.

Reply:
yes, I agree.